src/main/webapp/app/admin/statistics/statistics.component.ts

```diff
@@ -0,0 +1,165 @@
+import { Component, OnInit, OnChanges, ViewChild } from '@angular/core';
+import { StatisticsService } from 'app/admin/statistics/statistics.service';
+import { SPAN_PATTERN } from 'app/app.constants';
+import { ChartDataSets, ChartOptions, ChartType } from 'chart.js';
+import { BaseChartDirective, Label } from 'ng2-charts';
+import { DataSet } from 'app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component';
+import { TranslateService } from '@ngx-translate/core';
+import * as moment from 'moment';
+import { SpanType } from 'app/entities/statistics.model';
+
+@Component({
+    selector: 'jhi-statistics',
+    templateUrl: './statistics.component.html',
+})
+export class StatisticsComponent implements OnInit, OnChanges {
+    spanPattern = SPAN_PATTERN;
+    span: SpanType = SpanType.WEEK;
+
+    // Histogram related properties
+    public histogramData: number[] = [];
+    public barChartOptions: ChartOptions = {};
+    public barChartLabels: Label[] = [];
+    public barChartType: ChartType = 'bar';
+    public barChartLegend = true;
+    public UserLoginChartData: ChartDataSets[] = [];
+    public SubmissionsChartData: ChartDataSets[] = [];
+
+    @ViewChild(BaseChartDirective) chart: BaseChartDirective;
+
+    constructor(private service: StatisticsService, private translateService: TranslateService) {}
+
+    async ngOnInit() {
+        await this.setBinWidth();
+        await this.createChart();
+    }
+
+    async ngOnChanges() {}
+
+    private async setBinWidth(): Promise<void> {
+        switch (this.span) {
+            case SpanType.DAY:
+                this.histogramData = Array(24).fill(0);
+                break;
+            case SpanType.WEEK:
+                this.histogramData = Array(7).fill(0);
+                break;
+            case SpanType.MONTH:
+                this.histogramData = Array(moment().daysInMonth()).fill(0);
+                break;
+            case SpanType.YEAR:
+                this.histogramData = Array(12).fill(0);
+                break;
+        }
+    }
+
+    async getSubmissions(): Promise<number[]> {
+        return new Promise<number[]>((resolve, reject) => {
+            this.service.getTotalSubmissions(this.span).subscribe((res: number[]) => {
+                if (res !== null) {
+                    resolve(res);
+                } else {
+                    reject('Submissions could not get fetched');
+                }
+            });
+        });
+    }
+
+    onTabChanged(span: string): void {
```

Comment:
why are you not using the enum here?

Reply:
As we dont use Enums in the .html file, the method gets a String passed